src/borg/repository.py

```diff
@@ -662,17 +662,8 @@ def check_free_space(self):
             else:
                 # Keep one full worst-case segment free in non-append-only mode
                 required_free_space += full_segment_size
-        try:
-            st_vfs = os.statvfs(self.path)
-        except OSError as os_error:
-            logger.warning('Failed to check free space before committing: ' + str(os_error))
-            return
-        except AttributeError:
-            # TODO move the call to statvfs to platform
-            logger.warning('Failed to check free space before committing: no statvfs method available')
-            return
-        # f_bavail: even as root - don't touch the Federal Block Reserve!
-        free_space = st_vfs.f_bavail * st_vfs.f_frsize
+
+        free_space = get_free_space(self.path)
```

Comment:
thanks for moving this code.

Reply:
... but you need to keep the "return" semantics, so add a check like:
```
if free_space is None:  # None means it did not work.
    return
```
or alternatively, throw some own "did not work" exception and catch it here.